src/main/java/xyz/aadev/generitech/common/tileentities/machines/TileEntityFurnace.java

```diff
@@ -162,22 +169,35 @@ public float getTempRate() {
 
     @Override
     public void update() {
-        if (machineActive & !isSmeltPaused) {
+        if (container.getStoredPower()>=powerUsage && machineActive & !isSmeltPaused) {
             if (internalTemp < this.getMaxTemperature()) {
+                System.out.println("MaxTemperature");
                 internalTemp += getTempRate();
+                container.takePower(powerUsage,false);
             }
-        } else {
-            if (canIdle && internalTemp <= getIdleTemp()) {
-                internalTemp = getIdleTemp();
+        }else if (container.getStoredPower()>=powerUsage&&canIdle && internalTemp <= getIdleTemp())
+        {
+            System.out.println("getIdleTemp");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS106)